llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -171,6 +173,65 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
   return true;
 }
 
+/// Look for VINSERT sequence that can be rewritten as G_BUILD_VECTOR
+bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorInserts(
+    MachineInstr &MI, int SclSrcBits) const {
+  const Register DstReg = MI.getOperand(0).getReg();
+  LLT DstRegTy = MRI.getType(DstReg);
+  const LLT DstVecEltTy = DstRegTy.getElementType();
+  SmallVector<Register, 16> Regs;
```

Comment:
Could you move that definition closer to its place of use? I.e. around `// Collect registers in order for G_BUILD_VECTOR`